lib/HLSL/DxilValidation.cpp

```diff
@@ -3239,6 +3241,29 @@ static void ValidateFunctionBody(Function *F, ValidationContext &ValCtx) {
           unsigned opcode = OpcodeConst->getLimitedValue();
           DXIL::OpCode dxilOpcode = (DXIL::OpCode)opcode;
```

Comment:
This is undefined behavior if `opcode > NumOpCodes`. We should handle that case on the `unsigned` value before casting.

Reply:
Updated to unsigned compare.